File: AlgorithmsJS/BubbleSort.js
/**
 * Bubble Sort function O(N^2)
 */
function bubbleSort(list) {
  let unsortedUntilIndex = list.length
  let isSorted = false;

  while(!isSorted) {
    isSorted = true;
    for (let i = 0; i < unsortedUntilIndex; i++) {
      if (list[i] > list[i+1]) {
        isSorted = false;
        const lastNumber = list[i];
        list[i] = list[i + 1];
        list[i+1] = lastNumber;
      }
    }
    unsortedUntilIndex -=1;
  }

  return list;
}

module.exports = bubbleSort;

File: src/BubbleSort.kt
/**
 * Bubble Sort function O(N^2) - O(N)
 */
fun bubbleSort(list : Array<Int>) : Array<Int>{
    var unsortedUntilIndex : Int = list.size - 1
    var isSorted = false

    while(!isSorted) {
        isSorted = true
        for (i in 0 until unsortedUntilIndex) {
            if (list[i] > list[i + 1]) {
                isSorted = false
                val lastNumber = list[i]
                list[i] = list[i + 1]
                list[i + 1] = lastNumber
            }
        }
        unsortedUntilIndex -= 1
    }

    return list
}